src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.zaproxy.zap.extension.fuzz.payloads.ui.impl;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.GroupLayout;
+import javax.swing.*;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.fuzz.payloads.DefaultPayload;
+import org.zaproxy.zap.extension.fuzz.payloads.generator.NumberPayloadGenerator;
+import org.zaproxy.zap.extension.fuzz.payloads.generator.PayloadGenerator;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUI;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUIHandler;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUIPanel;
+import org.zaproxy.zap.model.MessageLocation;
+import org.zaproxy.zap.utils.ResettableAutoCloseableIterator;
+
+/**
+ *
+ * @author steven
+ */
+public class NumberPayloadGeneratorAdapterUIHandler implements
+        PayloadGeneratorUIHandler<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUI> {
+
+    private static final Logger LOGGER = Logger.getLogger(NumberPayloadGeneratorAdapterUIHandler.class);
+//    private static final String PAYLOAD_GENERATOR_NAME = Constant.messages.getString("fuzz.payloads.generator.number.name");
+    private static final String PAYLOAD_GENERATOR_NAME = "Numberzz";
+
+    @Override
+    public String getName() {
+        return PAYLOAD_GENERATOR_NAME;
+    }
+
+    @Override
+    public Class<NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUI> getPayloadGeneratorUIClass() {
+        return NumberPayloadGeneratorUI.class;
+    }
+
+    @Override
+    public Class<NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUIPanel> getPayloadGeneratorUIPanelClass() {
+        return NumberPayloadGeneratorUIPanel.class;
+    }
+
+    @Override
+    public PayloadGeneratorUIPanel<DefaultPayload, NumberPayloadGenerator, NumberPayloadGeneratorAdapterUIHandler.NumberPayloadGeneratorUI> createPanel() {
+        return new NumberPayloadGeneratorUIPanel();
+    }
+
+    public static class NumberPayloadGeneratorUI implements PayloadGeneratorUI<DefaultPayload, NumberPayloadGenerator> {
+
+        NumberPayloadGenerator numberPayloadGenerator;
+
+        public NumberPayloadGeneratorUI(NumberPayloadGenerator generator) {
+            this.numberPayloadGenerator = generator;
+        }
+
+        @Override
+        public Class<? extends NumberPayloadGenerator> getPayloadGeneratorClass() {
+            return NumberPayloadGenerator.class;
+        }
+
+        @Override
+        public String getName() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public String getDescription() {
+            return PAYLOAD_GENERATOR_NAME;
```

Comment:
While I kind of understand that a number generator should be somewhat self describing, it's probably worth being slightly more specific here. Consider `PAYLOAD_GENERATOR_DESC` with another i18n string.